### ISSUE TITLE
Not working on Android 9

### ISSUE THREAD

**Issue by sevicbb** [becvert/cordova-plugin-zeroconf#88]
```
zeroconf.watch('_http._tcp.', 'local.', ({ action, service }) => {
    console.log(service)
    if (action === "resolved") {
      console.log(service)
    } else if (action === "removed") {
      console.log(service)
    }
})
```

Same code works on iOS. It was working but suddenly it stopped. Any suggestions?

**Comment by emcniece** [comment]
The error line appears to be here: https://github.com/becvert/cordova-plugin-zeroconf/blob/master/src/android/net/becvert/cordova/ZeroConf.java#L555

```java
        Enumeration<String> names = service.getPropertyNames();
        while (names.hasMoreElements()) {
```

I suppose the `service.getPropertyNames()` method is returning an unexpected result. A solution might be just to try/catch this call. We should also look at `service` that gets passed into `jsonifyService(ServiceInfo service)` too,  it could be that service itself is empty and doesn't have any propertyNames to iterate. The mystery then becomes "what kind of service was returned from `registrationManager.register()` that is invalid?

Does Android 9 change or update app permission structures?

**Comment by SamuelMichelRiad** [comment]
Does anyone have a clue on how to resolve this? It's also not working on Android 11. Are there any alternative plugins to obtain local network devices?

**Comment by emcniece** [comment]
Yes, as per my previous comment,  https://github.com/becvert/cordova-plugin-zeroconf/blob/master/src/android/net/becvert/cordova/ZeroConf.java#L555 needs error handling and investigation. Apologies, I don't have capacity to do this at the moment :(
